Sources/StreamChatUI/StreamSwiftyMarkdown/SwiftyMarkdown/SwiftyScanner.swift

```diff
@@ -259,20 +259,27 @@ class SwiftyScanner {
 		} else {
 			var remainingTags = min(openRange.upperBound - openRange.lowerBound, closeRange.upperBound - closeRange.lowerBound) + 1
 			while remainingTags > 0 {
-				if remainingTags >= self.rule.maxTags {
+				let shouldAppendStyle = remainingTags >= self.rule.maxTags
+				if shouldAppendStyle {
 					remainingTags -= self.rule.maxTags
 					if let style = self.rule.styles[ self.rule.maxTags ] {
 						if !styles.contains(where: { $0.isEqualTo(style)}) {
 							styles.append(style)
 						}
 					}
 				}
-				if let style = self.rule.styles[remainingTags] {
+
+				let remainingTagsStyle = self.rule.styles[remainingTags]
```

Comment:
Is there a risk of out of bounds exception here?

Reply:
It is a dictionary